### PR TITLE
#320: hardcoded the size of MySQL data type TEXT

### LINKED ISSUE #320: `TEXT` type for MySQL dialect
## Situation

MySQL knows the type `TEXT` which can hold large amounts of text. Exasol's closest match is `VARCHAR`. At least documents up to 2 million characters should be usable through the MySQL dialect for Exasol Virtual Schema.

## Acceptance Criteria

* UTF text from MySQL `TEXT` columns visible as `VARCHAR` in VS (up to maximum Exasol `VARCHAR` size).

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #320

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dialects/mysql/MySqlColumnMetadataReader.java [hunk 1]
```diff
@@ -32,8 +35,34 @@ public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
...
     }
+
+    private DataType convertVarChar(final JdbcTypeDescription jdbcTypeDescription) {
+        final int size = getSize(jdbcTypeDescription);
+        final int octetLength = jdbcTypeDescription.getByteSize();
+        final DataType colType;
+        final DataType.ExaCharset charset = (octetLength == size) ? DataType.ExaCharset.ASCII
+                : DataType.ExaCharset.UTF8;
+        if (size <= DataType.MAX_EXASOL_VARCHAR_SIZE) {
+            final int precision = size == 0 ? DataType.MAX_EXASOL_VARCHAR_SIZE : size;
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/mysql/MySqlColumnMetadataReader.java]: Use a "if - else if" here for better readability.

**Comment by redcatbear** [comment]
Add handbook entry that describes the workaround and the problem in the MySQL driver behind it.

File: src/main/java/com/exasol/adapter/dialects/mysql/MySqlColumnMetadataReader.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/dialects/mysql/MySqlColumnMetadataReader.java]: done

File: doc/dialects/mysql.md [hunk 2]
```diff
@@ -34,7 +34,8 @@ It is necessary because JDBC driver requires a JAVA permission which we do not g
 1. [Create a bucket in BucketFS](https://docs.exasol.com/administration/on-premise/bucketfs/create_new_bucket_in_bucketfs_service.htm)
 1. Upload the driver to BucketFS
 
-This step is necessary since the UDF container the adapter runs in has no access to the JDBC drivers installed via EXAOperation but it can access BucketFS.
+This step is necessary since the UDF container the adapter runs in has no access 
+to the JDBC drivers installed via EXAOperation but it can access BucketFS.
```
Comment (redcatbear) [inline doc/dialects/mysql.md]: Why is there a newline here?

Comment (AnastasiiaSergienko) [reply doc/dialects/mysql.md]: fixed

**Pull request merged**